lib/protein/adapters/amqp_adapter/server.ex

```diff
@@ -91,8 +90,22 @@ defmodule Protein.AMQPAdapter.Server do
   defp init_conn_chan_queue(url, queue) do
     case Connection.open(url) do
       {:ok, conn} ->
-        {:ok, chan} = Channel.open(conn)
-        Queue.declare(chan, queue)
+        chan =
+          try do
+            {:ok, chan} = Channel.open(conn)
```

Comment:
Do we have to wrap both `Channel.open` and `Queue.declare` by the catch? I guess either one or the other throws the error, not both so it'd be better to only wrap what's needed to stay explicit and avoid unneeded catches.

Reply:
we don't, but it's symmetic with the `catch` clause below.

Reply:
That's not reason enough considering you risk catching too much and obfuscate the target function that actually throws. Please keep only the raising call within the try.